### DISCUSSION TITLE
AIO docker first run "/start.sh: line 313: exec: --: invalid option" how to solve?

### DISCUSSION THREAD

**Discussion by Termitebug** [nextcloud/all-in-one#3014] [Questions] [answered]
Hello, I would like to know why this error happens? Container errors itself at this stage and tries to restart indefinitely.
I used document setup command with --env data dir with correct permissions.
My machine is Debian 10 arm64 ROCK64.

**Comment by Zoey2936** [comment]
please show your docker run command

**Reply by Termitebug** [reply]
I used this docker command:

```
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
--env NEXTCLOUD_DATADIR="/mnt/ncdata/nextcloud" \
nextcloud/all-in-one:latest
```

**Comment by Zoey2936** [comment]
@szaimen as I think the docker CMD got overridden, maybe we should add the supervisor command directly to the start.sh instead of using it as cmd which get read by start.sh and then executes. since just doing `docker run ... nextcloud/all-in-one:latest abc` will execute abc instead of supervisor

**Reply by szaimen** [reply]
Yes, it looks like it. Can you propose a PR? :)

**Reply by Zoey2936** [reply]
I still obly have my smartphone, can you please?

**Reply by szaimen** [reply]
I am not completely sure what you wanted to do that is why I asked...

**Reply by Zoey2936** [reply]
instead of doing `exec` at the end just call supervisor -c conf and removing CMD from the dockerfile

**Reply by szaimen** [reply]
But will this break docker exec from working?

**Reply by Zoey2936** [reply]
why?

**Reply by szaimen** [reply, marked as answer]
Sorry, I understood it wrongly. PR is in https://github.com/nextcloud/all-in-one/pull/3018
